core/validation/validation.go

```diff
@@ -43,13 +47,18 @@ const spiffeRegexPrefixStart = "^"
 //
 //	bool: `true` if the SPIFFE ID belongs to VSecM Sentinel, `false` otherwise.
 func IsSentinel(spiffeid string) bool {
+	if !IsWorkload(spiffeid) {
+		return false
```

Comment:
Any VSecM entity is also a workload. So we run an `IsWorkload()` check before running other checks.